### PR TITLE
Add test if downloaded file exists after download

### PR DESCRIPTION
**Pull request by dithmer**
In this pull request, there is simply logic added to improve testing of downloading files into a directory.

### CODE REVIEW

File: chromedp_test.go [hunk 1]
```diff
@@ -642,30 +643,68 @@ func TestDownloadIntoDir(t *testing.T) {
 	ctx, cancel := testAllocate(t, "")
 	defer cancel()
 
+	// Wrap the context into a timeouted context, to ensure test with not run endless if download fails
+	// A second should be enough for this mocked download to finish, even on slower machines
+	ctx, cancel = context.WithTimeout(ctx, 1*time.Second)
+	defer cancel()
+
```
Comment (ZekeLu) [inline chromedp_test.go:650]: Let's remove this part. Let the go test tool handle test timeout.

File: chromedp_test.go [hunk 2]
```diff
@@ -642,30 +643,68 @@ func TestDownloadIntoDir(t *testing.T) {
...
+		}
+
+		fileContents, err := os.ReadFile(path)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if string(fileContents) != downloadContents {
+			t.Fatalf("Expected content of file to be '%s', but got '%s'", downloadContents, string(fileContents))
+		}
```
Comment (ZekeLu) [inline chromedp_test.go:706]: I think we could trust that the browser will save the file correctly. To simplify the test code, let's remove the checking of the file content.

**Review by ZekeLu — CHANGES_REQUESTED** [review]
Thank you!

File: chromedp_test.go (hunk 2 above)
Comment (dithmer) [reply chromedp_test.go]: Fully agree, changed that!

File: chromedp_test.go (hunk 1 above)
Comment (dithmer) [reply chromedp_test.go]: Alright, didn't know, this is something integrated into the tooling!

**Comment by dithmer** [comment]
Thank you for the feedback, I changed your hints. Anything else I can do here?

File: chromedp_test.go [hunk 3]
```diff
@@ -656,16 +657,38 @@ func TestDownloadIntoDir(t *testing.T) {
...
 		Click("#download", ByQuery),
 	); err != nil {
 		t.Fatal(err)
 	}
 
-	// TODO: wait for the download to finish, and check that the file is in
-	// the directory.
+	select {
+	case <-ctx.Done():
+		t.Fatal("chromedp context is done, before download was finished")
```
Comment (ZekeLu) [inline chromedp_test.go:680]: Replace this line with:

```go
t.Fatalf("unexpected error: %v", ctx.Err())
```

File: chromedp_test.go [hunk 4]
```diff
@@ -656,16 +657,38 @@ func TestDownloadIntoDir(t *testing.T) {
...
+	case guid := <-done:
+		path := filepath.Join(dir, guid)
+
+		if _, err := os.Stat(path); err != nil {
+			if errors.Is(err, os.ErrNotExist) {
+				t.Fatal("Downloaded file does not exist")
+			}
+
+			t.Fatal("Unknown error while checking downloaded file")
+		}
```
Comment (ZekeLu) [inline chromedp_test.go:690]: Replace the lines with:

```go
if _, err := os.Stat(filepath.Join(dir, guid)); err != nil {
	t.Fatal("want error nil, got: %v", err)
}
```

**Review by ZekeLu — CHANGES_REQUESTED** [review]
Thanks!

**Comment by ZekeLu** [comment]
Merged.

**Pull request closed without merging**